content/docs/conceptual-guides/branching.md

```diff
@@ -1,21 +1,56 @@
-- Run time travel queries against historical state
-- Run a set of queries with separate resources to avoid impacting your application
-- Tag and name the current moment for PITR convenience or ML model training repeatability
-- Run tests against a branch from production data
+A branch is a copy-on-write clone of your data. You can create a branch from a current or past state. For example, you can create a branch that includes all data up to the current point in time or up to a past point in time or [Log Sequence Number (LSN)](../../reference/glossary#lsn).
 
-_Note: The branch creation process does not increase load on the originating project. You can create a branch at any time without worrying about downtime or performance degradation._
+A branch is isolated from its originating data, so you are free to play around with it, modify it, and delete it when it's no longer needed. Changes to a branch are independent of the originating data. A branch and its parent share the same history but diverge at the point of branch creation. Writes to a branch are saved as an independent delta.
+
+Creating a branch does not increase load on the parent branch or affect it in any way, which means that you can create a branch at any time without impacting the performance of your production system.
+
+An endpoint is created for each branch, which is the compute instance associated with the branch. A branch endpoint is read-write. To connect to a branch from a client or application, you must connect to the branch endpoint.
+
+## Branching workflows
+
+You can use Neon's branching feature in variety development workflows, a few of which are discussed below.
+
+### Development
+
+You can create a branch of your production database that developers are free to play with and modify. You can quickly create a branch with all of the data that existed in the parent branch, eliminating the setup time required to deploy and maintain a development database. Branching is so easy and cost-effective that you can create a branch for each developer. For example, you can create branches from a primary development branch to assign tasks to be worked on in parallel.
+
+### Testing
+
+Branching enables testers to use the most recent production data. Testers can create branches for testing schema changes, validating new queries, or testing potentially destructive queries before deploying them into production. A branch is isolated from its parent branch but has all of the parent branch's data up to the point of branch creation, which eliminates the effort involved in hydrating a database. Tests can also run on separate branches in parallel, with each branch having dedicated compute resources.
+
+Another testing scenario enabled by branching is tracking down corruption or data quality issue. For example, you can create and dispose of as many point-in-time database branches as necessary to determine when the corruption or data quality issue first appeared.
```

Comment:
```suggestion
Another testing scenario enabled by branching is tracking down corruption or data quality issue. For example, you can create and dispose of multiple point-in-time database branches to determine when the corruption or data quality issue first appeared.
```

the amount of endpoints one can have simultaneously will be limited.
there is also a limit on the amount of branches you can have.